docs/topics/Gateway.md

```diff
@@ -135,7 +134,10 @@ Clients can detect zombied or failed connections by listening for [Opcode 11 Hea
 }
 ```
 
-If a client does not receive a heartbeat ack between its attempts at sending heartbeats, it should immediately terminate the connection with a non-1000 close code, reconnect, and attempt to resume.
+If a client does not receive a heartbeat ack between its attempts at sending heartbeats, this may be due to a failed or "zombied" connection. The client should then immediately terminate the connection with a non-1000 close code, reconnect, and attempt to [Resume](#DOCS_TOPICS_GATEWAY/resuming).
+
+> info
+> In the event of a service outage where you stay connected to the gateway, you should continue to heartbeat and receive ACKs. The gateway will eventually respond and issue a session once it's able to.
```

Comment:
Perhaps slightly unrelated, and maybe better fitting for another issue, but: Is there any recommendation for a minimum amount of time we should wait for a session to be issued? Or is it just, we should always wait an indefinite amount of time? I would presume an indefinite amount of time, but perhaps some clients may want to implement a timeout for this, for one reason or another.

Reply:
I believe the current advice is fine; as long as the gateway continues to heartbeat & ack, you can expect the service to be responsive. If you are otherwise required to reconnect, Discord will send you the appropriate close frame, Opcode 7 Reconnect, etc.

Clients are of course still free to implement a timeout in a hypothetical situation they send an `IDENTIFY`  and experience a very long delay before `READY`, but this is probably unwise; if the gateway is still issuing heartbeat acks, trying to start another session when Bigger Problems are afoot will continue to waste your session start limit. Timing out is probably undesirable from Discord's end as well if they expect clients to be patient as outlined above, for reliable load balancing.